utils/data-ingestor.js

```diff
@@ -0,0 +1,229 @@
+   * @param{Row} row - The row to operate on
+   * @returns{Row}
+   */
+  recomputeSpecialVars(row) {
+    let updatedRow = row;
+    const variable = updatedRow.get('variable');
+
+    try {
+      const specialType = updatedRow.get('specialType');
+
+      // do not recalculate values for 'normal' rows
+      if (specialType === undefined) return updatedRow;
+
+      const year = this.isPrevious ? PREV_YEAR : CUR_YEAR;
+      const { options } = find(specialCalcConfigs[this.profileType], ['variable', variable]);
+
+      updatedRow = this.recomputeSum(updatedRow, specialType, variable, year, options);
+      // decennial profiles do not have MOE values
+      if (this.profileType !== 'decennial') {
+        updatedRow = this.recomputeM(updatedRow, specialType, variable, year, options);
+      }
+    } catch (e) {
+      console.log(`Failed to update special vars for ${variable}:`, e); // eslint-disable-line
+    }
+
+    return updatedRow;
+  }
+
+  /*
+   * Recomputes the 'sum' value for a given row, applying the appropriate calculation
+   * based on the type of 'special' row. Additionally, scales the recomputed sum value
+   * if configured in 'options'
+   * @param{Row} row - The row to operate on
+   * @param{string} specialType - The type of this special variable
+   * @param{string} variable - The variable for the given row
+   * @param{string} year - The year for the given dataset
+   * @param{Object} options - Additional configuration for special calculations
+   * @returns{Row}
+   */
+  recomputeSum(row, specialType, variable, year, options) {
+    let updatedRow = row;
+    let sum;
+
+    if (specialType === 'median') {
+      const { trimmedEstimate, codingThreshold } = interpolate(this.data, variable, year);
+      sum = trimmedEstimate;
+      if (codingThreshold) {
+        updatedRow = updatedRow.set('codingThreshold', codingThreshold);
+        // if codingThreshold is set, indicates value was top- or bottom-coded,
+        // meaning the value is not reliable
+        updatedRow = updatedRow.set('is_reliable', false);
+      }
+    } else {
+      const formulaName = getFormulaName(options, 'sum');
+      sum = executeFormula(this.data, variable, formulaName, options.args);
+    }
+    sum = this.applyTransform(sum, options.transform);
+
+    return updatedRow.set('sum', sum);
+  }
+
+  /*
+   * Recomputes the 'm'(margin of error) value for a given row, applying the appropriate calculation
+   * based on the type of 'special' row. Additionally, scales the recomputed m value
+   * if configured in 'options'
+   * @param{Row} row - The row to operate on
+   * @param{string} specialType - The type of this special variable
+   * @param{string} variable - The variable for the given row
+   * @param{string} year - The year for the given dataset
+   * @param{Object} options - Additional configuration for special calculations
+   * @returns{Row}
+   */
+  recomputeM(row, specialType, variable, year, options) {
```

Comment:
I named a lot of variables poorly in the early iteration - should we find a better name for `recomputeM`? `recomputeSignificance`? (or is this reliability?).

Reply:
recomputeM is a function that recomputes the variable named 'm'... the variable named 'm' should be renamed, and that should be much easier now at least in the backend, but I thought that could wait for a separate PR